### PR TITLE
Backports from rails 5 branch

### PR DESCRIPTION
**Pull request by jhawthorn**
This PR has a number of backwards compatible changes from the rails 5 branch

### CODE REVIEW

File: api/spec/controllers/spree/api/orders_controller_spec.rb [hunk 1]
```diff
@@ -83,40 +73,41 @@ module Spree
...
-          and_return(true)
+      context "when the user cannot administer the order" do
+        stub_authorization! do |_|
+          can [:update], Spree::Order
+        end
 
-        allow(Spree::Order).to receive(:find_by!).
-          with(number: order.number).
-          and_return(order)
+        it "updates the cart contents" do
```
Comment (mamhoff) [inline api/spec/controllers/spree/api/orders_controller_spec.rb]: The spec checks for the order's email being changed, not for the cart contents. Huh?

File: api/spec/controllers/spree/api/orders_controller_spec.rb [hunk 2]
```diff
@@ -83,40 +73,41 @@ module Spree
...
 
         it "will associate users" do
-          expect(order).to receive(:associate_user!).
-            once.
-            with(user)
-
+          expect {
+            subject
+          }.to change { order.reload.user }.to(user)
           subject
```
Comment (mamhoff) [inline api/spec/controllers/spree/api/orders_controller_spec.rb]: I don't think this call to `subject` is necessary anymore, is it?

File: frontend/spec/controllers/spree/checkout_controller_spec.rb [hunk 3]
```diff
@@ -144,25 +144,29 @@ def post_address
...
-
-            post :update, {
-                state: "address",
-                order: {
-                    bill_address_attributes: order.bill_address.attributes.except("created_at", "updated_at"),
-                    ship_address_attributes: order.ship_address.attributes.except("created_at", "updated_at"),
-                    use_billing: false
-                }
+          subject do
+            -> {
```
Comment (mamhoff) [inline frontend/spec/controllers/spree/checkout_controller_spec.rb]: Is this lamda necessary? I think the `subject` call will already make the code in the block callable?

**Comment by mamhoff** [comment]
Module a bunch of nitpicks/questions, great work! 👍

**Comment by jhawthorn** [comment]
Thanks Martin. I fixed your comments. Good catches

**Comment by gmacdougall** [comment]
:+1:

Thanks John.

**Pull request merged**